dispatches/case_studies/nuclear_case/SynHist_integration.py

```diff
@@ -0,0 +1,132 @@
+#################################################################################
+# DISPATCHES was produced under the DOE Design Integration and Synthesis
+# Platform to Advance Tightly Coupled Hybrid Energy Systems program (DISPATCHES),
+# and is copyright (c) 2022 by the software owners: The Regents of the University
+# of California, through Lawrence Berkeley National Laboratory, National
+# Technology & Engineering Solutions of Sandia, LLC, Alliance for Sustainable
+# Energy, LLC, Battelle Energy Alliance, LLC, University of Notre Dame du Lac, et
+# al. All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. Both files are also available online at the URL:
+# "https://github.com/gmlc-dispatches/dispatches".
+#
+#################################################################################
+__author__ = "Gabriel J. Soto, Anna Wrobel"
+
+# This file contains utility functions for generating synthetic histories from a trained reduced
+#    order model from RAVEN.
+# Note that DISPATCHES, TEAL, and RAVEN are all assumed to be subdirectories within
+#    the same directory.
+import os
+import sys
+from os import path
+cwd = os.getcwd()
+proj_dir = path.dirname( path.abspath( path.join(cwd, '../../..') ) )
+raven_dir = path.abspath( path.join(proj_dir, 'raven') )
+sys.path.append( proj_dir )
+sys.path.append( raven_dir )
+sys.path.append( path.abspath( path.join(raven_dir, 'scripts')))
+sys.path.append( path.abspath( path.join(raven_dir, 'plugins')))
+
+from raven.ravenframework.utils import xmlUtils
+from raven.scripts import externalROMloader
+
+import numpy as np
+import operator
+
+###################
+class SynHist_integration():
+  """
+    Class to facilitate integration of RAVEN's Synthetic History generation.
+    Primary use is to load an already-trained price signal ARMA model. This model is
+    built during the constructor of this class.
+    Additional methods included for sampling from ARMA model and restructure data to accommodate
+    DISPATCHES workflow.
+  """
+
+  def __init__(self, target_file):
+    """
+      Constructor for SynHist_integration.
+      @ In, target_file, str, name of synthetic history file
+      @ Out, None
+    """
+    self.target_file = target_file # full path for ARMA file
+    self.inp = {'scaling': [1]} # this is a dummy input used for sampling call to ARMA model object
+    self.runner = self.buildRunner() # builds object to run ARMA model sampling
+
+  def buildRunner(self):
+    """
+      Builds synthetic history object from a pickledROM.
+      Currently only ARMA models are supported for pickledROM.
+    """
+    if not path.exists(self.target_file):
+      raise Exception(f"Target file not found at {self.target_file}")
+
+    runner = externalROMloader.ravenROMexternal(self.target_file, raven_dir)
```

Comment:
This should be switched over from `externalROMloader.ravenROMexternal` to `ROMExternal.ROMLoader`

Reply:
Fixed :)